### PR TITLE
Add support for <config.json>

### LINKED ISSUE #100: Add ability to format a list of file paths
I have a tool that exports DB objects and puts them in a repo and then formats them. Sometimes multiple objects are exported (say dependencies on a parent or multiples of a given object type) and passing a folder to be formatted isn't ideal (mostly because some things that weren't exported would get formatted).

I can go into a loop for each file, but that requires starting SQLcl which isn't the most efficient.

Perhaps adding the ability to pass in a list of file paths or maybe (more practically) a path to say a JSON file with such a list would help. Any thoughts?

### PR DESCRIPTION
**Pull request by dmcghan**
closes #100

### CODE REVIEW

**Comment by PhilippSalvisberg** [comment]
Thanks you, Dan. I created a branch based on this PR and then created a small test. It created the the following JSON file:

```json
[
    "/var/folders/lf/8g3r0ts900gfdfn2xxkn9yz00000gn/T/plsql-formatter-test-18221914857276644694/query.sql",
    "/var/folders/lf/8g3r0ts900gfdfn2xxkn9yz00000gn/T/plsql-formatter-test-18221914857276644694/markdown.md"
]
```

and then called `script format.js /private/var/folders/lf/8g3r0ts900gfdfn2xxkn9yz00000gn/T/plsql-formatter-test-18221914857276644694/config.json`.

The console output was:

```
Formatting file 1 of 2: /var/folders/lf/8g3r0ts900gfdfn2xxkn9yz00000gn/T/plsql-formatter-test-18221914857276644694/query.sql... done.
Formatting file 11 of 2: /var/folders/lf/8g3r0ts900gfdfn2xxkn9yz00000gn/T/plsql-formatter-test-18221914857276644694/markdown.md... done.
```

The files were processed (formatted) correctly, but the output `11 of 2` looks wrong. 

I changed line 390 as follows to fix the problem.

```
ctx.write("Formatting file " + parseInt(Number(i)+Number(1)) + " of " + files.length + ": " + files[i].toString() + "... ");
```

A more elegant solution is welcome.

I have some other issues with all existing SQLcl related tests. They are failing. I'm not sure what the problem is. I'll have to analyze it. 

Please update this PR when anytime. In the meantime I will write the test cases. They can be merged later into this PR.

**Comment by dmcghan** [comment]
> The files were processed (formatted) correctly, but the output 11 of 2 looks wrong.

I should have noticed that! :) I updated the code to make it a standard for loop.

**Comment by PhilippSalvisberg** [comment]
The current version does not work with rootPath anymore. That's why all other test cases fail.

File: sqlcl/format.js [hunk 1]
```diff
@@ -372,8 +409,12 @@ var run = function(args) {
         if (options.rootPath == "*") {
             formatBuffer(formatter);
         } else {
-            var files = getFiles(options.rootPath, options.extensions);
-            formatFiles(files, formatter, options.markdownExtensions);
+            if (options.files) {
```
Comment (PhilippSalvisberg) [inline sqlcl/format.js:412]: should be changed to `if (options.files.length > 0) {`

**Comment by dmcghan** [comment]
What are your thoughts on commit e278609? The idea is to move the 'imports' of the Java classes to the top (which is common in Node.js code, for example). That removed 30 lines of code from duplicate imports. 

Also, and perhaps a little more importantly for a JavaScript developer like myself, I prefixed the classes so one could tell where they came from and distinguish them from what's in JavaScript (e.g. String vs javaString).

File: sqlcl/format.js (hunk 1 above)
Comment (dmcghan) [reply sqlcl/format.js]: Good catch. Fixed.

**Comment by PhilippSalvisberg** [comment]
> What are your thoughts on commit [e278609](https://github.com/Trivadis/plsql-formatter-settings/commit/e2786099099f21fee8a80f1cd76f19191df44933)? The idea is to move the 'imports' of the Java classes to the top (which is common in Node.js code, for example). That removed 30 lines of code from duplicate imports.

It's probably better this way.

> Also, and perhaps a little more importantly for a JavaScript developer like myself, I prefixed the classes so one could tell where they came from and distinguish them from what's in JavaScript (e.g. String vs javaString).

I understand the reason for the prefix. But why not use a common prefix for Java classes. "java" for everything? The full qualified package is not that important IMO.

**Comment by dmcghan** [comment]
> I understand the reason for the prefix. But why not use a common prefix for Java classes. "java" for everything? The full qualified package is not that important IMO.

Done.

**Comment by PhilippSalvisberg** [comment]
I created a branch https://github.com/Trivadis/plsql-formatter-settings/tree/feature/pr-101-config-file based on your change plus a test case for a config file with a file array. All tests completed successfully. 

You can merge it into your branch.

I will add another test with all parameters later. I will probably need to change some other tests once you updated `printUsage`.

**Comment by dmcghan** [comment]
Okay, I've integrated the test changes and took the first stab at the doc. Let me know what you think.

**Comment by PhilippSalvisberg** [comment]
README.md looks good. Thanks a lot! 👍 

As soon as you merge my PR https://github.com/dmcghan/plsql-formatter-settings/pull/1 I can accept this PR.

**Comment by dmcghan** [comment]
Thanks, Philipp!

**Comment by PhilippSalvisberg** [comment]
I thank you, Dan. It was a pleasure working with you.

**Comment by rafael-trevisan** [comment]
Woohoo! Can I give this a test already? 😃

**Comment by PhilippSalvisberg** [comment]
Yes! Main Branch is the one to use with current SQLDev and SQLcl versions. A release is coupled with legacy versions. It's a different concept to other repos. So, the next release is planned when SQLDev 21.1 becomes available.

**Comment by rafael-trevisan** [comment]
It worked beautifully! Now I am generating a temp json file with my staged files within the pre-commit hook and it's perfect!

```shell
staged_files=$(git diff --cached --name-only --diff-filter=ACM | grep "db/")
jq -n --arg file "${staged_files}" '{ files: $file | split("\n") }' > temp/config.json
sqlcl ... script format.js temp/config.json
```

👏🏻

**Pull request merged**